### DISCUSSION TITLE
Airflow DAG serialization cache

### DISCUSSION THREAD

**Discussion by hgarrido** [apache/airflow#20162] [Q&A]
Hello,

I have deployed Airflow 2.2.2 on CeleryExecutor mode. I have built a DAG, which imports classes from a custom package. When I add a new attribute to this imported class, push a new version of my package, re-deploy Airflow with the correct new version package referenced (in a conda env), and add a reference to this new attribute in me DAG, Airflow UI shows an import error, stating that the newly added attribute does not exist in my class.

**What I have already tested:**

 - Hard-coding the attribute value in the DAG resulted in the DAG being correctly parsed
 - I verified that the correct version of my custom package was installed on the webserver/scheduler containers
 - Launching python3.7 on my webserver/scheduler containers (with the conda env activated), and importing the class from my custom package resulted in no  ```AttributeError```.
 - Re-deploying Airflow after my custom package deployment did not fix anything
 - Deleting all entries on `dag_code` and `serialized_dag` metedata tables did not fix the problem

**Conclusions so far**
It seems to me that this is related with the way Airflow 2.x handles dags serialization (https://airflow.apache.org/docs/apache-airflow/stable/dag-serialization.html), and that somehow the previous version of the imported class is still cached somewhere.
Any help on this would be much appreciated

**Airflow providers**
apache-airflow-providers-amazon==2.5.0
apache-airflow-providers-celery==2.1.0
apache-airflow-providers-ftp==2.0.1
apache-airflow-providers-http==2.0.1
apache-airflow-providers-imap==2.0.1
apache-airflow-providers-postgres==2.4.0
apache-airflow-providers-slack==4.1.0
apache-airflow-providers-sqlite==2.0.1
apache-airflow-providers-ssh==2.3.0

**Deployment**
Airflow 2.2.2 deployed on AWS ECS	

**Containers OS**
Debian GNU/Linux 10 (buster)

**Comment by potiuk** [comment]
Interesting. I think this is not the case, I believe just Airflow scheduler's environment is not properly initialized with your dependencies (missing paths, different user, dfferent behaviour with non-interactive way of starting the binary - and for example your .bashrc will not be sourced by Airflow. 

You can verify this by triggering reserilization (see https://github.com/apache/airflow/issues/19432 for details) - soon there will be a new airflow command to do just that but you can see some guidelines in the issue on how to do it manually.

My bet is that this will not fix the issue, but if it will then I think you should open an issue (after you work out some reproducible way of triggering it).
